Lib/test/test_sqlite3/test_cli.py

```diff
@@ -0,0 +1,155 @@
+        out = self.expect_success("-v")
+        self.assertIn(sqlite.sqlite_version, out)
+
+    def test_cli_execute_sql(self):
+        out = self.expect_success(":memory:", "select 1")
+        self.assertIn("(1,)", out)
+
+    def test_cli_execute_too_much_sql(self):
+        stderr = self.expect_failure(":memory:", "select 1; select 2")
+        err = "ProgrammingError: You can only execute one statement at a time"
+        self.assertIn(err, stderr)
+
+    def test_cli_execute_incomplete_sql(self):
+        stderr = self.expect_failure(":memory:", "sel")
+        self.assertIn("OperationalError (SQLITE_ERROR)", stderr)
+
+    def test_cli_on_disk_db(self):
+        out = self.expect_success(TESTFN, "create table t(t)")
+        self.addCleanup(unlink, TESTFN)
+        self.assertEqual(out, "")
+        out = self.expect_success(TESTFN, "select count(t) from t")
+        self.assertIn("(0,)", out)
+
+
+@requires_subprocess()
+class InteractiveSession(unittest.TestCase):
+    TIMEOUT = SHORT_TIMEOUT / 10.
+    MEMORY_DB_MSG = "Connected to a transient in-memory database"
+    PS1 = "sqlite> "
+    PS2 = "... "
+
+    def start_cli(self, *args):
+        return subprocess.Popen(
+            [sys.executable, "-m", "sqlite3", *args],
+            encoding="utf-8",
+            bufsize=0,
+            stdin=subprocess.PIPE,
+            # Note: the banner is printed to stderr, the prompt to stdout.
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+
+    def expect_success(self, proc):
+        proc.wait()
+        if proc.returncode:
+            self.fail("".join(proc.stderr))
+
+    def test_interact(self):
+        with self.start_cli() as proc:
+            out, err = proc.communicate(timeout=self.TIMEOUT)
+            self.assertIn(self.MEMORY_DB_MSG, err)
+            self.assertIn(self.PS1, out)
+            self.expect_success(proc)
+
+    def test_interact_quit(self):
+        with self.start_cli() as proc:
+            out, err = proc.communicate(input=".quit", timeout=self.TIMEOUT)
+            self.assertIn(self.MEMORY_DB_MSG, err)
+            self.assertIn(self.PS1, out)
+            self.expect_success(proc)
+
+    def test_interact_version(self):
+        with self.start_cli() as proc:
+            out, err = proc.communicate(input=".version", timeout=self.TIMEOUT)
+            self.assertIn(self.MEMORY_DB_MSG, err)
+            self.assertIn(sqlite.sqlite_version, out)
+            self.expect_success(proc)
+
+    def test_interact_valid_sql(self):
+        with self.start_cli() as proc:
+            out, err = proc.communicate(input="select 1;",
+                                        timeout=self.TIMEOUT)
+            self.assertIn(self.MEMORY_DB_MSG, err)
+            self.assertIn("(1,)", out)
+            self.expect_success(proc)
+
+    def test_interact_valid_multiline_sql(self):
+        with self.start_cli() as proc:
+            out, err = proc.communicate(input="select 1\n;",
```

Comment:
This may fail on Windows. Not sure if there's a platform independent way of doing this.